automation/taskcluster/rename_snapshot_artifacts.py

```diff
@@ -0,0 +1,126 @@
+"""
+Script to rename snapshots artifacts to a runtime centralized TIMESTAMP
+"""
+
+from __future__ import print_function
+
+import argparse
+import datetime
+import itertools
+import os
+import re
+import sys
+
+from decision_task import AAR_EXTENSIONS, HASH_EXTENSIONS
+
+
+SNAPSHOT_TIMESTAMP_REGEX = r'\d{8}\.\d{6}-\d{1}'
+
+
+def _extract_and_check_timestamps(archive_filename, regex):
+    """Function to check if a timestamp-based regex matches a given filename"""
+    match = re.search(regex, archive_filename)
+    try:
+        identifier = match.group()
+    except AttributeError:
+        raise Exception(
+            'File "{}" present in archive has invalid identifier. '
+            'Expected YYYYMMDD.HHMMSS-BUILDNUMBER within in'.format(
+                archive_filename
+            )
+        )
+    timestamp, buildno = identifier.split('-')
+    try:
+        datetime.datetime.strptime(timestamp, '%Y%m%d.%H%M%S')
+    except ValueError:
+        raise Exception(
+            'File "{}" present in archive has invalid timestamp. '
+            'Expected YYYYMMDD.HHMMSS within in'.format(
+                archive_filename
+            )
+        )
+
+    date, clock = timestamp.split('.')
+    return date, clock, buildno
+
+
+def _extract_old_timestamp(files):
+    """Function to find and ensure that a single timestamp has been used
+    across a component's related files"""
+    identifiers_collection = set()
+    for file_ in files:
+        (date, clock, bno) = _extract_and_check_timestamps(file_,
+                                                           SNAPSHOT_TIMESTAMP_REGEX)
+        _args = {
+            'date': date,
+            'clock': clock,
+            'buildno': bno
+        }
+        identifiers_collection.add(frozenset(_args.items()))
+
+    if len(identifiers_collection) > 1:
+        # bail if there are different timestamps/buildnumbers across the files
+        raise Exception('Different buildnumbers/timestamps identified within the files')
+    elif len(identifiers_collection) == 0:
+        # bail if no valid timestamp was found
+        raise Exception('No valid timestamp was identified within the files')
+
+    old_timestamp_dict = dict(next(iter(identifiers_collection)))
```

Comment:
Nit: This line can be simplified with `identifiers.pop()`

Reply:
👍